daemon/inspect.go

```diff
@@ -27,6 +28,23 @@ func (daemon *Daemon) ContainerInspect(ctx context.Context, name string, size bo
 		return daemon.containerInspectPre120(ctx, name)
 	case versions.Equal(version, "1.20"):
 		return daemon.containerInspect120(name)
+	case versions.LessThan(version, "1.44"):
+		ctr, err := daemon.ContainerInspectCurrent(ctx, name, size)
+		if err != nil {
+			return nil, err
+		}
+
+		shortCID := stringid.TruncateID(ctr.ID)
+		for _, ep := range ctr.NetworkSettings.Networks {
+			ep.Aliases = append(ep.Aliases, shortCID)
+			// Container hostname defaults to the container short ID if no hostname was specified -- we don't want
+			// duplicates.
+			if ctr.Config.Hostname != shortCID {
+				ep.Aliases = append(ep.Aliases, ctr.Config.Hostname)
+			}
+		}
```

Comment:
Double-checking here; this is adapting the response for API 1.29 - 1.43.  Is the container short-id to be found elsewhere in API 1.44? Are we still supporting that to be resolved (or was this only used as hostname _inside_ the container? i.e. related to the logic that was added specifically for that; https://github.com/moby/moby/blob/74cf9a0f60d5b91a36d3be3f9c0e3790cdd0e24a/pkg/stringid/stringid.go#L54-L59


- https://github.com/moby/moby/commit/9b7054fd0bc707f7f8e5c9ed9ada5a1664a19a34 /  https://github.com/moby/moby/pull/3881
- https://github.com/moby/moby/issues/3869

Reply:
> Is the container short-id to be found elsewhere in API 1.44?

I'm not sure why you're asking this question. I didn't check, but I don't see how any other short CID fields would relate to that change.

> Are we still supporting that to be resolved (or was this only used as hostname inside the container? i.e. related to the logic that was added specifically for that;

Yeah, the short CID is still a valid DNS name -- it can be resolved. So the logic in `stringid.GenerateRandomID()` is still relevant.

Reply:
> I'm not sure why you're asking this question. I didn't check, but I don't see how any other short CID fields would relate to that change.

Ah, yes, sorry, I honestly got lost in the weeds a bit with all the fields (and deprecated ones, and replacements for them).

I was wondering if there was a replacement for the information we're removing. In most (if not all) cases where we moved fields, the new ones were still providing the same information, and I wasn't sure if that was the case here (nor if there were use-cases for it).

Reply:
Right, so testing with this PR;

- in API 1.28 - 1.43, the `NetworkSettings.Networks["foo'}.Aliases` always contains the container's short-ID as alias
- in API 1.44, we're adding the new `DNSNames` field, but _ALSO_ removing the container's short-ID from the `NetworkSettings.Networks["foo'}.Aliases` field.

See:


```bash
docker network create mynetwork

docker run -d --name mycontainer1 --network mynetwork nginx:alpine
docker run -d --name mycontainer2 --network-alias myalias --network mynetwork nginx:alpine
```

### On API 1.43

```bash
curl -s --unix-socket /var/run/docker.sock http://localhost/v1.43/containers/mycontainer1/json | jq .NetworkSettings.Networks
{
  "mynetwork": {
    "IPAMConfig": null,
    "Links": null,
    "Aliases": [
      "2ba467ea502f"
    ],
    "MacAddress": "02:42:ac:13:00:02",
    "NetworkID": "68bea4575d8ee10a16151e90f58786f4afd62135fb5a2b412e16140b0ced48a5",
    "EndpointID": "e942106dbc637a66e770a8bd5bb79f61421ec3b8485ea2f1c186116427b9abc9",
    "Gateway": "172.19.0.1",
    "IPAddress": "172.19.0.2",
    "IPPrefixLen": 16,
    "IPv6Gateway": "",
    "GlobalIPv6Address": "",
    "GlobalIPv6PrefixLen": 0,
    "DriverOpts": null,
    "DNSNames": [
      "mycontainer1",
      "2ba467ea502f"
    ]
  }
}
curl -s --unix-socket /var/run/docker.sock http://localhost/v1.43/containers/mycontainer2/json | jq .NetworkSettings.Networks
{
  "mynetwork": {
    "IPAMConfig": null,
    "Links": null,
    "Aliases": [
      "myalias",
      "632ae41df334"
    ],
    "MacAddress": "02:42:ac:13:00:03",
    "NetworkID": "68bea4575d8ee10a16151e90f58786f4afd62135fb5a2b412e16140b0ced48a5",
    "EndpointID": "a0f4c5879225870e8c054a5af1c8e892e57a29557999dfbe9a577ac4eb988c2d",
    "Gateway": "172.19.0.1",
    "IPAddress": "172.19.0.3",
    "IPPrefixLen": 16,
    "IPv6Gateway": "",
    "GlobalIPv6Address": "",
    "GlobalIPv6PrefixLen": 0,
    "DriverOpts": null,
    "DNSNames": [
      "mycontainer2",
      "myalias",
      "632ae41df334"
    ]
  }
}
```

### On API 1.44

```
curl -s --unix-socket /var/run/docker.sock http://localhost/v1.44/containers/mycontainer1/json | jq .NetworkSettings.Networks
{
  "mynetwork": {
    "IPAMConfig": null,
    "Links": null,
    "Aliases": null,
    "MacAddress": "02:42:ac:13:00:02",
    "NetworkID": "68bea4575d8ee10a16151e90f58786f4afd62135fb5a2b412e16140b0ced48a5",
    "EndpointID": "e942106dbc637a66e770a8bd5bb79f61421ec3b8485ea2f1c186116427b9abc9",
    "Gateway": "172.19.0.1",
    "IPAddress": "172.19.0.2",
    "IPPrefixLen": 16,
    "IPv6Gateway": "",
    "GlobalIPv6Address": "",
    "GlobalIPv6PrefixLen": 0,
    "DriverOpts": null,
    "DNSNames": [
      "mycontainer1",
      "2ba467ea502f"
    ]
  }
}
curl -s --unix-socket /var/run/docker.sock http://localhost/v1.44/containers/mycontainer2/json | jq .NetworkSettings.Networks
{
  "mynetwork": {
    "IPAMConfig": null,
    "Links": null,
    "Aliases": [
      "myalias"
    ],
    "MacAddress": "02:42:ac:13:00:03",
    "NetworkID": "68bea4575d8ee10a16151e90f58786f4afd62135fb5a2b412e16140b0ced48a5",
    "EndpointID": "a0f4c5879225870e8c054a5af1c8e892e57a29557999dfbe9a577ac4eb988c2d",
    "Gateway": "172.19.0.1",
    "IPAddress": "172.19.0.3",
    "IPPrefixLen": 16,
    "IPv6Gateway": "",
    "GlobalIPv6Address": "",
    "GlobalIPv6PrefixLen": 0,
    "DriverOpts": null,
    "DNSNames": [
      "mycontainer2",
      "myalias",
      "632ae41df334"
    ]
  }
}
```

While that's not a change in the API _fields_, it _is_ a change in the meaning of an API field, which likely means we 

- must _deprecate_ that functionality (`Aliases` also contains the container's short-ID as network-scoped alias)
- keep the behavior for API 1.44 (which would return the short-ID _BOTH_ in `Aliases` _AND_ in `DNSNames`
- remove it for API 1.45

Reply:
Given that API 1.44 and below would _always_ have at least one entry in `Aliases` (it would never be `nil`?), perhaps we should also make sure that it always returns a slice (empty `[]` if there's no aliases) instead of `nil` 🤔